_data-prepper/pipelines/configuration/processors/grok.md

```diff
@@ -14,21 +14,34 @@ Takes unstructured data and utilizes pattern matching to structure and extract i
 
 Option | Required | Type | Description
 :--- | :--- | :--- | :---
-match | No | Map | Specifies which keys to match specific patterns against. Default is an empty body.
+match | No | Map | Specifies which keys to match specific patterns against. Default value is an empty body.
 keep_empty_captures | No | Boolean | Enables preserving `null` captures. Default value is `false`.
-named_captures_only | No | Boolean | enables whether to keep only named captures. Default value is `true`.
-break_on_match | No | Boolean | Specifies whether to match all patterns or stop once the first successful match is found. Default is `true`.
-keys_to_overwrite | No | List | Specifies which existing keys are to be overwritten if there is a capture with the same key value. Default is `[]`.
+named_captures_only | No | Boolean | Enables whether to keep only named captures. Default value is `true`.
+break_on_match | No | Boolean | Specifies whether to match all patterns or stop once the first successful match is found. Default value is `true`.
+keys_to_overwrite | No | List | Specifies which existing keys will be overwritten if there is a capture with the same key value. Default value is `[]`.
 pattern_definitions | No | Map | Allows for custom pattern use inline. Default value is an empty body.
 patterns_directories | No | List | Specifies the path of directories that contain customer pattern files. Default value is an empty list.
-pattern_files_glob | No | String | Specifies which pattern files to use from the directories specified for `pattern_directories`. Default is `*`.
-target_key | No | String | Specifies a parent level key to store all captures. Default value is `null`.
-timeout_millis | No | Integer | Maximum amount of time that should take place for the matching. Setting to `0` disables the timeout. Default value is `30,000`.
+pattern_files_glob | No | String | Specifies which pattern files to use from the directories specified for `pattern_directories`. Default value is `*`.
+target_key | No | String | Specifies a parent-level key to store all captures. Default value is `null`.
+timeout_millis | No | Integer | Maximum amount of time that takes place for matching. Setting to `0` disables the timeout. Default value is `30,000`.
 
 <!---## Configuration
 
-Content will be added to this section.
+Content will be added to this section.--->
 
 ## Metrics
 
-Content will be added to this section.--->
\ No newline at end of file
+<!---Tech: Is this block of text relevant/applicable here? "Apart from common metrics in [AbstractProcessor](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/processor/AbstractProcessor.java), the grok processor iontroduces the following custom metric."--->
```

Comment:
@graytaylor0 Is this block of text relevant/applicable here? "Apart from common metrics in [AbstractProcessor](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/processor/AbstractProcessor.java), the grok processor introduces the following custom metric."

This text is common in several of the processor documents Metrics sections, but isn't always included. If it is relevant/necessary, it is probably a good idea to include it.

Reply:
It's just a way to show the metrics that are common to all processors. It doesn't hurt to mention for all processor README's